cassandra-quarkus-extension/deployment/src/main/java/com/datastax/oss/quarkus/deployment/CassandraClientProcessor.java

```diff
@@ -0,0 +1,171 @@
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
+import io.quarkus.deployment.recording.RecorderContext;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+import io.quarkus.smallrye.health.deployment.spi.HealthBuildItem;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.Produces;
+
+class CassandraClientProcessor {
+  public static final String CASSANDRA_CLIENT = "cassandra-client";
+
+
+  @BuildStep
+  List<ReflectiveClassBuildItem> registerForReflection() {
+    return Arrays.asList(
+            new ReflectiveClassBuildItem(true, true, ExponentialReconnectionPolicy.class.getName()),
+            new ReflectiveClassBuildItem(true, true, NoopSchemaChangeListener.class.getName()),
+            new ReflectiveClassBuildItem(true, true, PassThroughAddressTranslator.class.getName()),
+            new ReflectiveClassBuildItem(true, true, DefaultLoadBalancingPolicy.class.getName()),
+            new ReflectiveClassBuildItem(true, true, DefaultRetryPolicy.class.getName()),
+            new ReflectiveClassBuildItem(true, true, NoSpeculativeExecutionPolicy.class.getName()),
+            new ReflectiveClassBuildItem(true, true, NoopNodeStateListener.class.getName()),
+            new ReflectiveClassBuildItem(true, true, NoopRequestTracker.class.getName()),
+            new ReflectiveClassBuildItem(true, true, PassThroughRequestThrottler.class.getName()),
+            new ReflectiveClassBuildItem(true, true, AtomicTimestampGenerator.class.getName()));
+  }
+
+
+
+  @SuppressWarnings("unchecked")
+  @Record(STATIC_INIT)
+  @BuildStep
+  BeanContainerListenerBuildItem build(
+      RecorderContext recorderContext,
+      CassandraClientRecorder recorder,
+      BuildProducer<FeatureBuildItem> feature,
+      BuildProducer<GeneratedBeanBuildItem> generatedBean) {
+
+    feature.produce(new FeatureBuildItem(CASSANDRA_CLIENT));
+
+    String cassandraClientProducerClassName = getCassandraClientProducerClassName();
+    createCassandraClientProducerBean(generatedBean, cassandraClientProducerClassName);
+
+    return new BeanContainerListenerBuildItem(
+        recorder.addCassandraClient(
+            (Class<? extends AbstractCassandraClientProducer>)
+                recorderContext.classProxy(cassandraClientProducerClassName)));
+  }
+
+  private String getCassandraClientProducerClassName() {
+    return AbstractCassandraClientProducer.class.getPackage().getName()
+        + "."
+        + "CassandraClientProducer";
+  }
+
+  private void createCassandraClientProducerBean(
+      BuildProducer<GeneratedBeanBuildItem> generatedBean,
+      String cassandraClientProducerClassName) {
```

Comment:
Yes, I wanted to simplify that, but the code is required because of the initialization order.
Please note the usage of `CassandraClientRecorder recorder,`. It is creating a proxy class that extends this auto-generated code. I was unable to make it work the way you suggested.

Reply:
FWIW I got this to work (where "work" means "the tests, or at least the non-native ones but including the integration tests, are passing") via the following:

https://github.com/absurdfarce/cassandra-quarkus/pull/1

Obviously that impl is extremely basic and not very idiomatic... I was largely just trying to demonstrate the idea.

Reply:
Yes, I was also thinking about loading this class using IO but I think that it is a bit fragile. 
When I was analyzing other extensions in the `quarkus` project, none of them is loading class this way. They tend to generate it. 

One important note here, when I was debugging it, it seems that quarkus is not creating a physical file with `CassandraClientProducer` class - you will not find this file in any `/target` directory. It seems that the generated content is kept only in memory (and discarded later)

Reply:
I am creating a new ticket for possible improvement of this code in the future:
https://datastax-oss.atlassian.net/browse/JAVA-2722

Reply:
To be clear: I'm not necessarily saying it's a good idea to solve this problem in this way.  It very well _might_ be but that isn't immediately clear to me.  I _could_ see the pattern of providing a build item based on an existing class file as being something which could possibly be re-used.  And while it's true that the file in question isn't being created at deployment time it shouldn't be an issue since it's already contained in the extension itself.